src/test/kotlin/com/hibob/unittests/UserServiceTest.kt

```diff
@@ -0,0 +1,86 @@
+package com.hibob.unittests
+
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.mockito.Mockito.*
+
+class UserServiceTest{
+
+    private val userDao = mock<UserDao>()
+    private val notificationService = mock<NotificationService>()
+    private val emailVerificationService = mock<EmailVerificationService>()
+    private val userService = UserService(userDao, notificationService, emailVerificationService)
+
+    @Test
+    fun `Registering a user who is registered already`() {
+        val user = User(1L, "Noy", "noy.robinson@hibob.io", "123456", true)
+        whenever(userDao.findById(1L)).thenReturn(user)
+        assertThrows<IllegalArgumentException> { userService.registerUser(user) }
+    }
+
+    @Test
+    fun `User registration fails`() {
+        val user = User(1L, "Noy", "noy.robinson@hibob.io", "123456", false)
+        whenever(userDao.findById(1L)).thenReturn(null)
+        whenever(userDao.save(user)).thenReturn(false)
+        assertThrows<IllegalStateException> { userService.registerUser(user) }
```

Comment:
assertThrows returns the exception, so you can also test that the exception message is correct

Reply:
I'll add it :)